processing/src/main/java/io/druid/segment/virtual/SingleStringInputDimensionSelector.java

```diff
@@ -153,7 +153,7 @@ public IdLookup idLookup()
   @Override
   public Object getObject()
   {
-    return defaultGetObject();
+    return lookupName(getRow().get(0));
```

Comment:
I'm not sure this will work for empty rows in multi-value dimensions.

Reply:
Right, fixed. Thanks.